src/optimizer/query_planner.c

```diff
@@ -1005,8 +1005,27 @@ qo_top_plan_new (QO_PLAN * plan)
 			  orderby_skip = pt_sort_spec_cover (plan->iscan_sort_list, order_by);
 
 			  /* try using a reverse scan */
-			  if (!orderby_skip)
+			  if (orderby_skip)
 			    {
+			      /* CBRD-24636
+			       *
+			       * orderby_skip is true because the column order of the index used for the scan
+			       * and the column order of order by are the same without reverse order.
+			       * So the PT_HINT_USE_IDX_DESC hint should not be applied.
+			       */
+
+			      PT_HINT_ENUM *hint = &(tree->info.query.q.select.hint);
+			      bool has_hint = (*hint & PT_HINT_USE_IDX_DESC) > 0;
+
+			      if (has_hint)
+				{
+				  *hint = (PT_HINT_ENUM) (*hint & ~PT_HINT_USE_IDX_DESC);
```

Comment:
Equivalant with `tree->info.query.q.select.hint = (PT_HINT_ENUM) (tree->info.query.q.select.hint & ~PT_HINT_USE_IDX_DESC);`

Reply:
Is the intention to reduce the use of temporary variables?

Reply:
Maybe defining macro functions to perform bitwise operations for the PT_HINT_ENUM type would be better. But it is trivial. the current implementation seems enough.